edb/schema/indexes.py

```diff
@@ -74,131 +74,21 @@ def is_index_valid_for_type(
     # HACK: currently this helper just hardcodes the permitted index & type
     # combinations, but this should be inferred based on index definition.
 
-    # TODO: remove the explicit hardcode of types compatible with indexes and
-    # make them configurable in the schema instead. This is especially
-    # important for extension types.
-    #
+    # TODO:
```

Comment:
Can you clean up issubclass Optional bit now?

Reply:
Cleaned up